src/rpc/types.rs

```diff
@@ -11,23 +11,19 @@ use serde_this_or_that::as_u64;
 #[derive(Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[allow(missing_docs)]
 pub struct ExecutionPayloadValidation {
-    pub parent_hash: H256,
+    pub parent_hash: B256,
     pub fee_recipient: Address,
-    pub state_root: H256,
-    pub receipts_root: H256,
+    pub state_root: B256,
+    pub receipts_root: B256,
     pub logs_bloom: Bloom,
-    pub prev_randao: H256,
-    #[serde(deserialize_with = "as_u64")]
-    pub block_number: u64,
-    #[serde(deserialize_with = "as_u64")]
-    pub gas_limit: u64,
-    #[serde(deserialize_with = "as_u64")]
-    pub gas_used: u64,
-    #[serde(deserialize_with = "as_u64")]
-    pub timestamp: u64,
+    pub prev_randao: B256,
```

Comment:
The reason why we had the custom serialization above is because we are currently still aiming to be consistent with the current flashbots builder api. (See: https://github.com/ultrasoundmoney/reth-payload-validator/pull/8)

In their api numbers are encoded as decimal strings whereas the default deserialization of reth U64 type is hex. Therefore these changes would break the api.

I suggest removing these changes and limiting the PR to only what is necessary to make it compatible with the updated reth dependency.

Reply:
Note that this referes to the u64 fields only. The change from H256 to B256 is fine imo.